### DISCUSSION TITLE
Nightly builds for Windows

### DISCUSSION THREAD

**Discussion by deevus** [zed-industries/zed#15159] [Show and tell]
If you wanted to test Zed on Windows, I have created an automated nightly build here: https://github.com/deevus/zed-windows-builds

Quickest way to install it is via [Scoop](https://scoop.sh/):

```
scoop bucket add versions
scoop install versions/zed-nightly
```

**Comment by mapsgeek** [comment]
Thanks

**Comment by Curious-r** [comment]
Great work bro!

**Comment by cdcb52** [comment]
Granted its not technically released stable for Windows. I am having an issue where I attempt to run it and nothing ever opens. No error messages are displayed. Anyone have any thoughts on the cause/solution?

**Reply by vikrant0017** [reply]
Yes I faced this issue. Turned out it was issue related with VulkanSDK . My laptop has a integrated AMD Graphics and discrete NVIDIA card. Turns out I didn't have AMD driver installed. Installing it fixed the issue for me. 
This may not work for everyone, but you could give it a try.

**Reply by lhutyra** [reply]
I experienced exactly sam eproblem - nothing appear, nothign happend on click 'zed' - It was outdated integradted(iris) graphic card driver

**Reply by co-bby** [reply]
> I experienced exactly sam eproblem - nothing appear, nothign happend on click 'zed' - It was outdated integradted(iris) graphic card driver

have you been able to resolve. I have the same issue

**Reply by cdcb52** [reply]
> > I experienced exactly sam eproblem - nothing appear, nothign happend on click 'zed' - It was outdated integradted(iris) graphic card driver
> 
> 
> 
> have you been able to resolve. I have the same issue

I updated my iris driver to the newest version on windows 10. Didn't modify the install of zed and it works now. Seemed like it was vulkan related. I had similar issues on my arch Linux install also. Updated vulkan driver and worked.

**Comment by Master-Y0da** [comment]
can be updated ?...I hit "check for updates" button in zed options, but no answer.

**Reply by deevus** [reply]
Probably not until there are official builds for Windows.

**Reply by cdcb52** [reply]
For how I installed I can not upgrade from the app but rather recompile with the latest bin

**Reply by deevus** [reply]
If you have installed using `scoop`, updates can be attained by running `scoop update zed`. 

You can automate this using [UniGetUI](https://github.com/marticliment/UnigetUI)

**Reply by aus10d** [reply]
> If you have installed using `scoop`, updates can be attained by running `scoop update zed`.

for me, to update it, i had to specify "zed-nightly"

**Comment by meetgovindbajaj** [comment]
i m trying to run it in win 11, using latest intel arc driver version, no external graphic card is there with i5 11gen, all drivers and software and win versions are latest, still not able to see the app running. i checked task manager and found that its open in background but not able to access or open it though.

**Reply by lhutyra** [reply]
Same here - latests build not starts

**Reply by JunkuiZhang** [reply]
This is an issue introduced by a recent commit, #18705 will fix this.

**Reply by yodatak** [reply]
For me build before 1 october works and latest don't work

**Reply by lhutyra** [reply]
@yodatak  - I think it's not merged yet

**Reply by co-bby** [reply]
> For me build before 1 october works and latest don't work

you can try the latest build here https://github.com/shenjackyuanjie/zed-win-build/releases/tag/0.156.0%2B85eceb7194-1.2.1

**Reply by lhutyra** [reply]
For me latest nightly works already. #18705 merged

**Reply by meetgovindbajaj** [reply]
yes its opening now

**Reply by aus10d** [reply]
after updating, mine now works too! very happy to try it on windows! thank you, @deevus, for making this for us!

**Comment by Nickeldon** [comment]
<img src="https://i.ytimg.com/vi/pEBSp-fkNog/maxresdefault.jpg"/>

**Comment by lhutyra** [comment]
Why I cannot download latest version? For me last one is 7th November...

**Reply by codesbyayush** [reply]
builds after 7th november are failing for some reason so 7th november was the last successful release

**Comment by b1rigu** [comment]
OMG, thanks for this. It took me 4 hours to build on version on my laptop

**Comment by DeanLemans** [comment]
can you please fix the windows build?
there hasnt been an update to the [releases](https://github.com/deevus/zed-windows-builds/releases) for 2 weeks.

**Comment by luckzylp** [comment]
+1

**Comment by deevus** [comment]
Apologies for the delay. My email address is on my profile if I become unresponsive.

**Comment by Deku-nattsu** [comment]
for some reason when i build it myself it doesn't come with a lot of things such as project panel, and built in terminal

**Comment by Jimmy-Z** [comment]
Thank you!

And, is there plan to support tagged builds?

**Reply by deevus** [reply]
I don't personally need them, but if someone wanted to make a PR, I would consider it.

**Comment by max-programming** [comment]
Wow thanks a lot buddy!

**Comment by MElkmeshi** [comment]
it doesn't open for me

**Reply by MElkmeshi** [reply]
![image](https://github.com/user-attachments/assets/ce7bec8a-7687-4d79-b63d-41c76732b0eb)

**Comment by Nichokas** [comment]
On this builds the colab work?

and btw thanks for making a builld system

**Comment by ARyaskov** [comment]
@deevus Thank you very much for GH Actions workflow and Scoop release!
Sadly it (ver. 20250117) does not work for me today on Windows 10 with the latest Windows updates and 3 years old PC hardware with integrated Intel's GPU, latest VC++ redist and WebView2 (nor "zed-opengl.exe", nor "zed.exe", nor "scoop-version of zed" -- it just doesn't not appears in Task Manager, no windows, no error messages).
Hope I'll find a way to use it on Windows without building :)

**Reply by deevus** [reply]
Try installing vcredist2022 via scoop

**Reply by ARyaskov** [reply]
Thanks for advise @deevus ! But it still doesn't run as previously :(

**Comment by weebao** [comment]
Just wanna say thanks this is awesome. Does anyone have any tip on how to set up github copilot on this? I already signed in but nothing shows up for github copilot

**Comment by jloa** [comment]
```
Error: Cannot find module 'D:\C:\Users\julius\AppData\Local\Zed\extensions\work\vue\node_modules\@vue\language-server\bin\vue-language-server.js'
```

Why does nightly doesnt have the fix for this issue?
Also weird that zed doesnt respect the settings

```
 "lsp": {
    "vue": {
      "binary": {
        "path": "C:/@vue/language-server/bin/vue-language-server.js"
      },
    }
  }
```

**Reply by deevus** [reply]
The fix for that hasn't hit `main` yet. They aren't prioritising PRs for Windows at this time.

**Reply by jloa** [reply]
Anyway to change the path via config somehow?

**Reply by itpropro** [reply]
This also impacts basically every extension I am using, guess we just have to wait.

**Comment by pigletfly** [comment]
does this build support wsl ?

**Reply by deevus** [reply]
In what sense? I'm pretty sure it can open files that are in WSL.

**Reply by pigletfly** [reply]
> In what sense? I'm pretty sure it can open files that are in WSL.

open files from wsl directly, like vscode behaviors, just type code.

**Reply by deevus** [reply]
I'm not sure. You'd have to test it and report back.

**Reply by codeit-ninja** [reply]
Ofcourse it does not support WSL itself. But since WSL is, as its name says, a subsystem. Which means you can access its filesystem from its host. Its basically a VM. On windows it shows WSL folders under network.

**Reply by RenaudLN** [reply]
One can open files that are on WSL but what about getting access to the environments installed on WSL? All my dev stuff is setup on the Linux side with Python virtual environments etc. When opening things naively Zed still tries to use Windows for LSP etc.

**Reply by deevus** [reply]
I don't think that kind of workflow is properly supported yet.

**Reply by airtonix** [reply]
use ssh and setup the linux remote server?

**Reply by airtonix** [reply]
scratch that, latest zed editor now supports wsl. 

1. launch zed on windows
2. `ctrl + shift + alt + o` > add wsl distro
3. keep following prompts. 
4. 🥳

**Comment by rzkyyds** [comment]
Tried it on windows 11 but the app doesn't show up when opened.

**Reply by rzkyyds** [reply]
![{CE6E884D-D428-47DA-8AB2-FDCB49611169}](https://github.com/user-attachments/assets/1d9133ad-4b53-4967-b375-9cfe693154c3)

**Reply by deevus** [reply]
I've had this happen to me before. Try opening using the Start Menu shortcut.

**Reply by rzkyyds** [reply]
I'm trying but it's still not opening the App

**Reply by wanjohiryan** [reply]
Same here, Zed does not open on Windows 11

**Reply by cxinmayy** [reply]
> I'm trying but it's still not opening the App

did you find any solution??

**Reply by rzkyyds** [reply]
still can't open until now, i guess i have to wait until there is a stable build from zed directly.

**Comment by kireerik** [comment]
Installer doesn't start on Windows ARM.

**Reply by deevus** [reply]
I'm not currently shipping ARM builds, as I have no way to test it. 

If you're willing to make a PR to add an ARM build, I'd accept it.

**Comment by snehilrshah** [comment]
Hey, I am not able to update zed-nightly after the 2025_03_09 build

![image](https://github.com/user-attachments/assets/2ac704f5-7739-45f2-be22-f0468277a630)

**Reply by deevus** [reply]
See here: https://github.com/deevus/zed-windows-builds/issues/41

**Comment by mohamad-tohidi** [comment]
thank you very much

**Comment by weebao** [comment]
@deevus quick question is this automatically updated for every new zed update? thanks!

**Reply by deevus** [reply]
Yep. It updates daily. Stable releases are usually only a couple of times a week.

**Comment by fxrstor** [comment]
@deevus Starts and almost instantly exits in Windows 22H2 despite having `vcredist2022` installed in the computer. Both stable and nightly builds don't work. 
```ps
> Invoke-RestMethod -Uri https://get.scoop.sh | Invoke-Expression
Initializing...
Downloading...
Creating shim...
Adding ~\scoop\shims to your path.
Scoop was installed successfully!
Type 'scoop help' for instructions.
> scoop bucket add extras
Checking repo... OK
The extras bucket was added successfully.
> scoop install extras/zed
Installing 'zed' (0.181.5) [64bit] from 'extras' bucket
zed.zip (79.5 MB) [====================================================================================================================================================================================] 100%
Checking hash of zed.zip ... ok.
Extracting zed.zip ... done.
Linking ~\scoop\apps\zed\current => ~\scoop\apps\zed\0.181.5
Creating shim for 'zed'.
Making C:\Users\InsertUsername\scoop\shims\zed.exe a GUI binary.
Creating shortcut for Zed (zed.exe)
'zed' (0.181.5) was installed successfully!

> zed
> scoop install vcredist2022
Installing 'vcredist2022' (14.42.34438.0) [64bit] from 'extras' bucket
vc_redist.x64.exe (24.5 MB) [=================================================================================] 100%
Checking hash of vc_redist.x64.exe ... ok.
vc_redist.x86.exe (13.3 MB) [=================================================================================] 100%
Checking hash of vc_redist.x86.exe ... ok.
Linking ~\scoop\apps\vcredist2022\current => ~\scoop\apps\vcredist2022\14.42.34438.0
Running post_install script...WARN  This product is already installed
WARN  This product is already installed
done.
'vcredist2022' (14.42.34438.0) was installed successfully!
Notes
-----
You can now remove this installer with 'scoop uninstall vcredist2022'
> scoop install extras/zed-opengl
Installing 'zed-opengl' (0.181.5) [64bit] from 'extras' bucket
zed-opengl.zip (79.4 MB) [====================================================================================] 100%
Checking hash of zed-opengl.zip ... ok.
Extracting zed-opengl.zip ... done.
Linking ~\scoop\apps\zed-opengl\current => ~\scoop\apps\zed-opengl\0.181.5
Creating shim for 'zed'.
WARN  Overwriting shim ('zed.exe' -> 'zed-opengl.exe') installed from zed
Making C:\Users\InsertUsername\scoop\shims\zed.exe a GUI binary.
Creating shortcut for Zed (zed-opengl.exe)
'zed-opengl' (0.181.5) was installed successfully!
```

**Reply by deevus** [reply]
Try installing latest GPU drivers for your system.

**Comment by weebao** [comment]
Do you guys have a problem with changing models in the assistant model after opening the settings file once?

**Reply by patyhank** [reply]
Same here. After opening the settings file, any changes won’t be saved, even if I close the file tab.
Exiting the application and reopening Zed makes it work properly again.

**Comment by essenciary** [comment]
Used to work, it stopped working a few weeks ago. Updated multiple times, no dice. The app just doesn't start (I presume it crashes upon startup). Automatically installed and updated via Scoop. 

OS: Windows 11
Build: 24H2 (26100)
Uptime: 0 days, 1 hour, 54 minutes
Resolution: 1920x1200 @60Hz
CPU: AMD Ryzen 5 8640HS w/ Radeon 760M Graphics     
GPU: AMD Radeon 760M Graphics
Memory: 15624 MB

**Reply by essenciary** [reply]
Crazy, but found this in the docs and fixed it. 

> If there's nothing Vulkan-related in the logs and you happen to have Bandicam installed, try uninstalling it. Zed is currently not compatible with Bandicam.

**Comment by jackytank** [comment]
thanks brother

**Comment by SameerJS6** [comment]
## Unofficial Zed CLI for Windows

If anyone here is looking for a CLI on Windows, you can try this: [Zed CLI for Windows (Unofficial)](https://github.com/SameerJS6/zed-cli-win-unofficial). It’s written in Go, simple to use, and can also add an “Open with Zed” option to the context menu with just a single command.

You can download the CLI using either PowerShell scripts or Scoop—whichever method you prefer.

### Powershell Script

#### Install CLI 
Download and run the installation script for the Unofficial Zed CLI:

```powershell
irm https://raw.githubusercontent.com/SameerJS6/zed-cli-win-unofficial/refs/heads/main/scripts/release/install-wrapper.ps1 | iex
```

#### Install Zed + CLI (All-in-One)
Install both Zed (Unofficial Build from @pirafrank) and the Unofficial CLI with zero setup. This script handles everything automatically:

```powershell
irm https://raw.githubusercontent.com/SameerJS6/zed-cli-win-unofficial/refs/heads/main/scripts/release/install-with-zed-wrapper.ps1 | iex
```

### Scoop

#### 1. Add the bucket
```powershell
scoop bucket add zed-cli-unofficial https://github.com/SameerJS6/zed-cli-win-unofficial
```

#### 2. Install the CLI
```powershell
scoop install zed-cli-unofficial/zed-cli-win-unofficial
```

### Command for Installing "Open with Zed" Feature 
```powershell 
zed context install
```

![main-website-mockup](https://github.com/user-attachments/assets/24895015-1599-454e-bc34-f41111c7a602)

**Comment by Neuron-Mr-White** [comment]
Thank you, damn!!!

**Comment by franciskafyi** [comment]
Doing some spring cleaning and stumbled on this one 😅 we have official support for Windows since last year: https://zed.dev/download. Closing this 🫡
